### PR TITLE
🌱 Improve SSA patch test

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
This is just a first step to incrementally improve this test. There is more to come because we probably found an issue in apiserver, but I"ll open a separate issue for that.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
/assign @fabriziopandini @chrischdi 

More to come

**Review by chrischdi — COMMENTED** [review]
/lgtm

File: internal/util/ssa/patch_test.go [hunk 1]
```diff
@@ -161,6 +147,11 @@ func TestPatch(t *testing.T) {
 		// Verify that request was not cached (as it changed the object)
 		g.Expect(ssaCache.Has(requestIdentifier)).To(BeFalse())
 
+		// Wait for 1 second. We are also trying to verify in this test that the resourceVersion of the Machine
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply internal/util/ssa/patch_test.go:150]: only one nit (not blocking): what about linking the corresponding API Server issue?

[the remark this answers is not in the document]
Comment (sbueringer) [reply internal/util/ssa/patch_test.go:150]: Sure!

[the remark this answers is not in the document]
Comment (sbueringer) [reply internal/util/ssa/patch_test.go:150]: Linked both issues (now that there is a CAPI issue as well)

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**